sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -380,12 +381,20 @@ private[hive] class SparkSQLCLIDriver extends CliDriver with Logging {
 
           ret = rc.getResponseCode
           if (ret != 0) {
-            // For analysis exception, only the error is printed out to the console.
-            rc.getException() match {
-              case e : AnalysisException =>
-                err.println(s"""Error in query: ${e.getMessage}""")
+            rc.getException match {
+              case e: AnalysisException => e.cause match {
+                case Some(_) =>
+                  err.println(
+                    s"""Error in query: ${e.getMessage}
+                       |${org.apache.hadoop.util.StringUtils.stringifyException(e)}
+                     """.stripMargin)
```

Comment:
Printing this stacktrace via logger (with `ERROR` levels) is not enough for your usecase? I feel that printing to a shell is a bit verbose.

Reply:
For AnalysisException, we won't get the stacktrace via logger. It's in the `DEBUG` level

Reply:
FYI, https://github.com/apache/spark/blob/1fac06c4307c8c7a5a48a50952d48ee5b9ebccb2/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala#L71

Reply:
Or do you mean just use `logError`?

Reply:
Ah, I see. But, I'm still not sure that end users want to see stacktraces of analysis exceptions for debugging use. Could we add a new internal config to turn on/off this error info in a server-side?

Reply:
Just for clarification, although this class is under the `thriftserver` package, but it has nothing to do with the `thriftserver` and there is no such `client/server` thing here. It is only a single `spark-sql` shell process. 

For the config thing,  maybe we just need `sessionState.getIsSilent` and just like

```scala
case e: AnalysisException => e.cause match {
                case Some(_) if !sessionState.getIsSilent =>
                  err.println(
                    s"""Error in query: ${e.getMessage}
                       |${org.apache.hadoop.util.StringUtils.stringifyException(e)}
                     """.stripMargin)
             case ....
```

Reply:
Yea, I personally think the proposed one above looks better than that in the current PR.